dataplane/vppagent/pkg/vppagent/vppagent.go

```diff
@@ -323,11 +325,37 @@ func (v *VPPAgent) Init(common *common.DataplaneConfigBase, monitor *crossconnec
 	v.setDataplaneConfigVPPAgent(monitor)
 	v.reset()
 	v.programMgmtInterface()
-	v.metricsCollector = metrics.NewMetricsCollector()
-	v.metricsCollector.CollectAsync(monitor, v.vppAgentEndpoint)
+	v.setupMetricsCollector(monitor)
 	return nil
 }
 
+func (v *VPPAgent) setupMetricsCollector(monitor *crossconnect_monitor.CrossConnectMonitor) {
+	val, ok := os.LookupEnv(DataplaneMetricsCollectorEnabledKey)
+	if ok {
+		enabled, err := strconv.ParseBool(val)
+		if err != nil {
+			logrus.Errorf("Metrics collector is not using, %v ", err)
```

Comment:
I suppose we need to enable it in this case.

Reply:
fixed